CRM/Core/Form.php

```diff
@@ -1375,7 +1379,7 @@ public function addSelect($name, $props = array(), $required = FALSE) {
    * @throws \Exception
    * @return HTML_QuickForm_Element
    */
-  public function addField($name, $props = array(), $required = FALSE, $legacyDate = TRUE) {
+  public function addField($name, $props = array(), $required = FALSE, $fieldSpec = array(), $legacyDate = TRUE) {
```

Comment:
@colemanw what is your opinion on this. It feels like the easiest way to make the fields on the payment processor form metadata driven. However, some of the fields are pseudofields so I figured maybe we could change the scope here to allow them to be passed in